spec/models/validations/sales/property_validations_spec.rb

```diff
@@ -48,4 +48,40 @@
       end
     end
   end
+
+  describe "#validate_property_unit_type" do
+    context "when number of bedrooms is <= 1" do
+      let(:record) { FactoryBot.build(:sales_log, beds: 1, proptype: 2) }
+
+      it "does not add an error if it's a bedsit" do
+        property_validator.validate_bedsit_number_of_beds(record)
+        expect(record.errors).not_to be_present
+      end
+    end
+
+    context "when number of bedrooms is > 1" do
+      let(:record) { FactoryBot.build(:sales_log, beds: 2, proptype: 2) }
+
+      it "does add an error if it's a bedsit" do
+        property_validator.validate_bedsit_number_of_beds(record)
+        expect(record.errors.added?(:proptype, "Bedsit maximum 1 bedroom")).to be true
+        expect(record.errors.added?(:beds, "Bedsit bedroom maximum 1")).to be true
+      end
+
+      it "does not add an error if proptype is undefined" do
```

Comment:
@arfacamble you wrote a comment about potentially making these tests more comprehensive by checking for when one of the fields is undefined. What do you think about these (this test and the one below)?

Reply:
looks great